src/main/java/Main.java

```diff
@@ -1,31 +1,58 @@
 import java.util.HashMap;
-
+ 
 public class Main {
-  
+  private static CLIParser parser;
+  private static final int DEFAULT_PORT_NUMBER = 7777;
+ 
   public static void main(String[] args) {
     try {
-      HashMap<String, Handler> routesMap = createRoutesMap();
-      Router router = new Router(routesMap);
-      
-      int port = Integer.parseInt(args[0]);
+      parser = new CLIParser(args);
+
+      DataStore store = extractDataStore();
+      Handler defaultHandler = new FileHandler(store);
+ 
+      Router router = setUpRouter(defaultHandler);
+      int port = extractPortNumber();
       Server server = new Server(port, router);
       server.start();
-    } catch (ArrayIndexOutOfBoundsException e) {
-      System.err.println("Plese enter a port number as a command line argument.");
-      System.err.println("Usage: java HTTPServer <port>");
+    } catch (NonexistentDirectoryException e) {
+      System.err.println(e.getMessage());
+      System.err.println("Please enter a valid directory from which to serve content.");
+    } catch (NullPointerException e) {
+      System.err.println("You must provide a data store to start the server.");
+      System.err.println("Usage: java -jar http-server.jar -port <port> -<store-flag> <store argument>");
+      printAvailableStores();
     } catch (NumberFormatException e) {
-      System.err.println("Argument must be a number.");
-    } 
+      System.err.println("Port must be a number.");
+    }
   }
 
-  public static HashMap<String, Handler> createRoutesMap() {
+  private static DataStore extractDataStore() throws NonexistentDirectoryException {
+    String publicDirectoryPath = parser.getDirectory();
+    return new Directory(publicDirectoryPath);
+  }
```

Comment:
Looks like you're assuming that the data store will be a directory path, which is confusing since the method is called `extractDataStore`

Reply:
I've updated `CLIParser` with a `getStoreFlag` and a List of possible store flags. `extractDataStore` in `Main` can now easily extend to handle new kinds of stores.